Fix gimbal pointing backwards

Jaeyoung-Lim opened the pull request:
This PR fixes https://github.com/PX4/Firmware/issues/14902, where the gimbal was pointing backwards. 

@julianoes @TSC21 It seems like the problem is that we transform the yaw command, and not the actual gimbal orientation to the correct frame. While this is not a "fix", nevertheless I would like to have this fix in so that the end user does not suffer from the issue.

I will try to fix it properly in the meantime

TSC21 commented on src/gazebo_gimbal_controller_plugin.cpp:
```diff
@@ -483,7 +483,7 @@ void GimbalControllerPlugin::OnUpdate()
 
     // We want yaw to control in body frame, not in global.
-    double yaw_command = this->yawCommand + this->lastImuYaw;
+    double yaw_command = this->yawCommand + this->lastImuYaw + M_PI;
```
This still doesn't seem well explained.

Jaeyoung-Lim replied:
I think the fix should be done by transforming the gimbal orientation in the bodyframe and not by adding the orientation to the command itself. I will try to fix it properly, but this is something that we can hide the issue from the end user

TSC21 replied:
Right. So why not use the body frame orientation quaternion directly? You can create a quaternion of the yaw command, apply the rotation and extract yaw.

Jaeyoung-Lim replied:
Ah, right. That might be easier. I was trying to keep the yaw command and make everything else in bodyframe

Jaeyoung-Lim committed:
Translate yaw to point at the right direction

The typhoon h480 gimbal was pointing in the wrong direction. 

This was because the vehicle orientation was being read from the imu orientation. This causes a problem when the imu orientation does not match the vehicle orientation. This commit fixes the problem by using the groundtruth orientation instead

Jaeyoung-Lim committed:
Cleanup
